pants-plugins/macros.py

```diff
@@ -112,6 +112,65 @@ def st2_component_python_distribution(**kwargs):
     python_distribution(**kwargs)  # noqa: F821
 
 
+# Default copied from PEX (which uses zipfile standard MS-DOS epoch).
+# https://github.com/pex-tool/pex/blob/v2.1.137/pex/common.py#L39-L45
+MTIME = "1980-01-01T00:00:00Z"
+
+# These are used for system packages (rpm/deb)
+ST2_PACKS_GROUP = "st2packs"
+ST2_SVC_USER = "st2"
+
+
+def st2_pack_archive(**kwargs):
+    """Create a makeself_archive using files from the given dependencies.
+
+    This macro should be used in the same BUILD file as the pack_metadata target.
+    """
+    build_file_path = build_file_dir()  # noqa: F821
+    if "st2tests" == build_file_path.parts[0]:
+        # avoid creating duplicate archive for the core pack
+        # which is also located under st2tests/st2tests/fixtures/packs
+        return
+    pack_name = build_file_path.name  # noqa: F821
+
+    dependencies = kwargs.pop("dependencies", [])
+    if ":metadata" not in dependencies:
+        dependencies = [":metadata", *dependencies]
+
+    # This is basically a "wrap_as_files" target (which does not exist yet)
+    shell_command(  # noqa: F821
+        name="files",
+        execution_dependencies=dependencies,
+        command="true",
+        output_directories=["."],
+        root_output_directory=".",
+    )
+
+    # https://www.pantsbuild.org/stable/docs/shell/self-extractable-archives
+    # https://www.pantsbuild.org/stable/reference/targets/makeself_archive
+    makeself_archive(  # noqa: F821
+        name="archive",
+        label=f"{pack_name} StackStorm pack",
+        files=[
+            ":files",  # archive contents
+            "//:license",  # LICENSE file included in archive header, excluded from contents
+        ],
+        args=(  # see: https://makeself.io/#usage
+            # Makeself expects '--arg value' (space) not '--arg=value' (equals) for cmdline
+            "--license",
+            "__archive/LICENSE",
+            "--target",
+            f"/opt/stackstorm/packs/{pack_name}",
+            # reproducibility flags:
+            "--tar-extra",  # extra tar args: '--arg=value' (equals delimited) space separated
+            f"--owner=root --group={ST2_PACKS_GROUP} --mtime={MTIME} --exclude=LICENSE",
```

Comment:
```suggestion
            f"--owner=root --group={ST2_PACKS_GROUP} --mtime={MTIME}",
```

Should the LICENSE file be installed with the pack like this? If we install it, it will end up as `/opt/StackStorm/packs/<pack>/LICENSE`.

Reply:
The pack licence is that of st2, so there's no need to install a copy in each pack directory in my opinion.